### PR TITLE
Added `databricks labs ucx ensure-assessment-run` to CLI commands.

### LINKED ISSUE #674: Add `databricks labs ucx ensure-assessment-run` command
`databricks labs ucx ensure-assessment-run --workspace-profiles profile-one,profile-two,profile-three` to check if assessment workflows did run otherwise start them.

Upstream dependencies:
- https://github.com/databrickslabs/ucx/issues/673

Pick up the state from https://github.com/databrickslabs/ucx/blob/main/src/databricks/labs/ucx/install.py#L1041-L1058

### PR DESCRIPTION
**Pull request by FastLee**
Closes #674 
Description:
This change adds a CLI command that performs the following logic.

1.  Check whether UCX is installed for the current user on the workspace
2. Check whether the Assessment workflow was successfully run or is still running
3. Wait for a current run to complete or start a new run and wait for it to complete.

### CODE REVIEW

File: labs.yml [hunk 1]
```diff
@@ -43,3 +43,10 @@ commands:
 
   - name: create-table-mapping
     description: create initial table mapping for review
+
+  - name: ensure-assessment-run
+    description: ensure the assessment job was run on a workspace
+    flags:
+      - name: run
+        description: run the assessment workflow if it hasn't been successfully run
```
Comment (nfx) [inline labs.yml:51]: let's run it by default if it's not there, no need for an option

File: src/databricks/labs/ucx/cli.py [hunk 2]
```diff
@@ -48,17 +61,11 @@ def skip(schema: str, table: str | None = None):
         return None
     ws = WorkspaceClient()
     installation_manager = WorkspaceInstaller(ws)
-    logger.info("Fetching installation config.")
```
Comment (nfx) [inline src/databricks/labs/ucx/cli.py:51]: don't refactor the functions, for now. also - use the proper InstallationManager (and not the WorkspaceInstaller)

https://github.com/databrickslabs/ucx/blob/main/src/databricks/labs/ucx/cli.py#L84-L87

File: src/databricks/labs/ucx/cli.py [hunk 3]
```diff
@@ -90,13 +97,28 @@ def create_table_mapping():
     webbrowser.open(f"{ws.config.host}/#workspace{path}")
 
 
+def ensure_assessment_run(*, run: bool = False):
+    ws = WorkspaceClient()
+    installation_manager = WorkspaceInstaller(ws)
```
Comment (nfx) [inline src/databricks/labs/ucx/cli.py:102]: this is not an installation manager.

File: src/databricks/labs/ucx/cli.py [hunk 4]
```diff
@@ -90,13 +97,28 @@ def create_table_mapping():
...
+    config = get_config(installation_manager)
+    if not config:
+        return None
+    if run:
+        installation_manager.validate_and_run("assessment")
```
Comment (nfx) [inline src/databricks/labs/ucx/cli.py:107]: let's do only "validate_and_run"

File: src/databricks/labs/ucx/install.py [hunk 5]
```diff
@@ -860,6 +861,19 @@ def _remove_install_folder(self):
         except InvalidParameterValue:
             logger.error("Error deleting install folder")
 
+    def validate_step(self, step: str) -> bool:
+        job_id = self._state.jobs[step]
+        logger.debug(f"Validating {step} workflow: {self._ws.config.host}#job/{job_id}")
+        completed_runs = list(self._ws.jobs.list_runs(completed_only=True, job_id=job_id))
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:867]: what if the assessment run is in progress? please add a test case as well

**Review by nfx — CHANGES_REQUESTED** [review]

File: labs.yml (hunk 1 above)
Comment (FastLee) [reply labs.yml:51]: Done

File: src/databricks/labs/ucx/install.py (hunk 5 above)
Comment (FastLee) [reply src/databricks/labs/ucx/install.py]: Done

File: src/databricks/labs/ucx/cli.py (hunk 3 above)
Comment (FastLee) [reply src/databricks/labs/ucx/cli.py]: Fixed

**Review by nfx — APPROVED** [review]
please make PR changelog-ready

**Pull request merged**